deps/verifier/src/tdx/eventlog.rs

```diff
@@ -67,21 +85,73 @@ impl CcEventLog {
     }
 
     pub fn query_digest(&self, entity: MeasuredEntity) -> Option<String> {
-        let event_desc_prefix = Self::generate_query_key_prefix(entity)?;
-
         for event_entry in self.cc_events.log.clone() {
-            if event_entry.event_desc.len() < event_desc_prefix.len() {
-                continue;
-            }
-            if &event_entry.event_desc[..event_desc_prefix.len()] == event_desc_prefix.as_slice() {
-                let digest = &event_entry.digests[0].digest;
-                return Some(hex::encode(digest));
+            match (entity, event_entry.event_type) {
+                (MeasuredEntity::TdvfKernel, ev)
+                    if ev.as_str() == "EV_EFI_BOOT_SERVICES_APPLICATION" =>
```

Comment:
Can you put this into the match itself?

something like 
```rust
match (entity, event_entry.event_type.as_str())
    (MeasuredEntity::TdvfKernel, "EV_EFI_BOOT_SERVICES_APPLICATION") => {
```

Reply:
much better, thanks